src/hb-ot-cmap-table.hh

```diff
@@ -1497,7 +1499,19 @@ struct cmap
 
       this->get_glyph_data = subtable;
       if (unlikely (symbol))
-	this->get_glyph_funcZ = get_glyph_from_symbol<CmapSubtable>;
+      {
+	switch ((unsigned) face->table.OS2->get_font_page ()) {
+	default:
+	  this->get_glyph_funcZ = get_glyph_from_symbol<CmapSubtable>;
+	  break;
+	case OS2::font_page_t::FONT_PAGE_SIMP_ARABIC:
+	  this->get_glyph_funcZ = get_glyph_from_symbol_arabic_pua1<CmapSubtable>;
+	  break;
+	case OS2::font_page_t::FONT_PAGE_TRAD_ARABIC:
+	  this->get_glyph_funcZ = get_glyph_from_symbol_arabic_pua2<CmapSubtable>;
+	  break;
+	}
+      }
       else
```

Comment:
So this currently works only for `ot` font functions (I was surprised again to discover that `ft `are still the default for the command line tools, it confused me more than once). I don’t fell easy about this since it is unlikely that clients implementing their own font functions will try to support this, we also need to support it for the `ft` functions, not sure what is the best way, though.